test/trento/domain/host/host_test.exs

```diff
@@ -651,6 +651,33 @@ defmodule Trento.HostTest do
   end
 
   describe "deregistration" do
+    test "should reject all the commands except the registration ones when the host is deregistered" do
+      host_id = Faker.UUID.v4()
+      dat = DateTime.utc_now()
+
+      initial_events = [
+        build(:host_registered_event, host_id: host_id),
+        %HostDeregistered{
+          host_id: host_id,
+          deregistered_at: dat
+        }
+      ]
+
+      commands_to_reject = [
+        %DeregisterHost{},
+        %RequestHostDeregistration{},
+        %RollUpHost{},
+        %UpdateHeartbeat{},
+        %UpdateProvider{},
+        %UpdateSlesSubscriptions{},
+        %HostDetailsUpdated{}
```

Comment:
This is wrong, `HostDetailsUpdated` is an event, not a command.
You should have `RegisterHost`, and you would need to update it after the restoration code.
Up to you, you can remove that entry and work on it during in the restoration changes